oak_dice/src/utils.rs

```diff
@@ -29,3 +33,16 @@ pub fn cbor_encoded_bytes_to_vec(bytes: &[u8]) -> Result<Vec<u8>, String> {
         .map_err(|err| format!("failed to write bytes: {:?}", err))?;
     Ok(result)
 }
+
+/// Like the slice `copy_from_slice` method but does not panic if slices are not
+/// the same length. In the case of a shorter source slice, only overwrites
+/// the beginning of the destination slice. In the case of a longer source slice
+/// it throws an error.
+pub(crate) fn padded_copy_from_slice(dst: &mut [u8], src: &[u8]) -> Result<(), String> {
```

Comment:
The order of the arguments seems a bit counterintuitive. I imagine you did it this way to match copy_from_slice, but that's a bit misleading since that has a self receiver and usually called as a method on a slice, while this must always be called as a stand alone function. Anyways up to you

Reply:
reimplemented this as a trait on `[u8]`, which should avoid any confusion. :)